src/plugins/matrixops/diag_operation.cpp

```diff
@@ -81,41 +75,6 @@ namespace phylanx { namespace execution_tree { namespace primitives
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    primitive_argument_type diag_operation::diag0d(
-        primitive_argument_type&& arg, std::int64_t k) const
-    {
-        switch (extract_common_type(arg))
-        {
-        case node_data_type_bool:
-            return diag0d(
-                extract_boolean_value_strict(std::move(arg), name_, codename_),
-                k);
-
-        case node_data_type_int64:
-            return diag0d(
-                extract_integer_value_strict(std::move(arg), name_, codename_),
-                k);
-
-        case node_data_type_double:
-            return diag0d(
-                extract_numeric_value_strict(std::move(arg), name_, codename_),
-                k);
-
-        case node_data_type_unknown:
-            return diag0d(
-                extract_numeric_value(std::move(arg), name_, codename_), k);
-
-        default:
-            break;
-        }
-
-        HPX_THROW_EXCEPTION(hpx::bad_parameter,
-            "diag_operation::diag0d",
-            generate_error_message(
-                "the diag primitive requires for all arguments to "
-                    "be numeric data types"));
-    }
-
```

Comment:
What's the rationale to remove `diag` for scalars? Is this change related to `diag_d`?

Reply:
Because Numpy diag does not consider scalar case. To comply with it, I delete the diag0d case.